packages/core/src/routes/interaction/types/index.ts

```diff
@@ -0,0 +1,34 @@
+export type Identifier =
+  | AccountIdIdentifier
+  | VerifiedEmailIdentifier
+  | VerifiedPhoneIdentifier
+  | SocialIdentifier;
+
+export type AccountIdIdentifier = { key: 'accountId'; value: string };
+
+export type VerifiedEmailIdentifier = { key: 'verifiedEmail'; value: string };
+
+export type VerifiedPhoneIdentifier = { key: 'verifiedPhone'; value: string };
+
+export type SocialIdentifier = { key: string; value: UseInfo };
```

Comment:
i would suggest to use `key: 'social'` and put another field to identify the social type/id?

Reply:
updated